src/main/java/org/dice_research/opal/civet/metrics/DescriptionMetricImproved.java

```diff
@@ -0,0 +1,161 @@
+package org.dice_research.opal.civet.metrics;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.vocabulary.DCTerms;
+import org.dice_research.opal.civet.Metric;
+import org.dice_research.opal.common.vocabulary.Opal;
+import opennlp.tools.postag.POSModel;
+import opennlp.tools.postag.POSSample;
+import opennlp.tools.postag.POSTaggerME;
+import opennlp.tools.tokenize.WhitespaceTokenizer;
+
+/**
+ * The Description metric - Improved version checks the RDF property
+ * dct:description. If the dct:description contains more number nouns, verbs and
+ * adjectives then high stars are awarded, and less are awarded in a descending
+ * order.
+ * 
+ * Rating Criteria is as follows: "If dct:description contains <5 nouns, verbs
+ * or adjectives then 1 star" + "else if dct:description contains >5 and <10
+ * nouns, verbs or adjectives then 2 stars" + "else if dct:description contains
+ * >10 and <15 nouns, verbs or adjectives then 3 stars" + "else if
+ * dct:description contains >15 and <20 nouns, verbs or adjectives then 4 stars"
+ * + "else if dct:description contains >20 nouns, verbs or adjectives then 5
+ * stars"
+ * 
+ * @see:"https://opennlp.apache.org/docs/1.9.0/manual/opennlp.html"
+ * @see:"http://csjarchive.cogsci.rpi.edu/Proceedings/2013/papers/0263/paper0263.pdf"
+ * @see:"https://pressbooks.bccampus.ca/technicalwriting/chapter/importanceverbs/"
+ * @see:"https://www.diva-portal.org/smash/get/diva2:231227/fulltext01.pdf"
+ * 
+ * @author Aamir Mohammed
+ */
+
+public class DescriptionMetricImproved implements Metric {
+
+	private static final String DESCRIPTION = "If dct:description contains <5 nouns, verbs or adjectives then 1 star"
+			+ "else if dct:description contains >5 and <10 nouns, verbs or adjectives then 2 stars"
+			+ "else if dct:description contains >10 and <15 nouns, verbs or adjectives then 3 stars"
+			+ "else if dct:description contains >15 and <20 nouns, verbs or adjectives then 4 stars"
+			+ "else if dct:description contains >20 nouns, verbs or adjectives then 5 stars";
+
+	// German parts of speech tags for noun, verb and adjective
+	private final String nounTag = "NN";
+	private final String verbTag = "VVFIN";
+	private final String adjectiveTag = "ADJA";
+	int countNouns = 0;
+	int lastIndexNoun = 0;
+	int countVerbs = 0;
+	int lastIndexVerb = 0;
+	int countAdjectives = 0;
+	int lastIndexAdjective = 0;
+
+	public int posTagger(String description) throws IOException {
+
+		// Loading Parts of speech-maxent model
+		InputStream inputStream = new FileInputStream("src/main/resources/de-pos-maxent.bin");
```

Comment:
What is this file? How does it work? Could please put the source or any reference for this file in this card?
https://trello.com/c/E9oiV7IR/80-description-metric-improvement